### PR TITLE
refac: energy supplier on child metering points

### PR DESCRIPTION
**Pull request by johevemi**

### CODE REVIEW

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py [hunk 1]
```diff
@@ -35,24 +35,35 @@ def get_metering_points_and_child_metering_points(
 
     es = "energy_supplier_id_temp"
     mp = "metering_point_id_temp"
+    fd = "from_date_temp"
+    td = "to_date_temp"
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:39]: Can we lower the cognitive load by not introducing uncommon abbrevations?
```suggestion
    from_date = "from_date_temp"
    to_date = "to_date_temp"
```

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py [hunk 2]
```diff
@@ -35,24 +35,35 @@ def get_metering_points_and_child_metering_points(
...
+            == all_metering_points[Colname.parent_metering_point_id]
+        )
+        & (
+            production_and_consumption_metering_points[fd]
+            >= all_metering_points[Colname.from_date]
+        )
+        & (
+            production_and_consumption_metering_points[fd]
+            < all_metering_points[Colname.to_date]
+        ),  # parent_metering_point_id is always null on child metering points
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:64]: Did you mean never null?
Consider also moving this comment to the relevant place (line 55-ish). Should

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py [hunk 3]
```diff
@@ -35,24 +35,35 @@ def get_metering_points_and_child_metering_points(
...
         production_and_consumption_metering_points.withColumnRenamed(
             Colname.energy_supplier_id, es
-        ).withColumnRenamed(Colname.metering_point_id, mp)
+        )
+        .withColumnRenamed(Colname.metering_point_id, mp)
+        .withColumnRenamed(Colname.from_date, fd)
+        .withColumnRenamed(Colname.to_date, td)
     )
 
     all_metering_points = _get_all_child_metering_points(metering_point_periods_df)
```
[the remark this answers is not in the document]
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:49]: fixed

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:64]: removed

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py [hunk 4]
```diff
@@ -35,61 +35,132 @@ def get_metering_points_and_child_metering_points(
 
     es = "energy_supplier_id_temp"
     mp = "metering_point_id_temp"
-    production_and_consumption_metering_points = (
-        production_and_consumption_metering_points.withColumnRenamed(
-            Colname.energy_supplier_id, es
-        ).withColumnRenamed(Colname.metering_point_id, mp)
+    from_date = "from_date_temp"
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:38]: I like that you changed from fd to from_date.

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py [hunk 5]
```diff
@@ -35,61 +35,132 @@ def get_metering_points_and_child_metering_points(
...
+        all_child_metering_points[Colname.settlement_method],
+        all_child_metering_points[Colname.grid_area],
+        all_child_metering_points[Colname.resolution],
+        all_child_metering_points[Colname.from_grid_area],
+        all_child_metering_points[Colname.to_grid_area],
+        all_child_metering_points[Colname.parent_metering_point_id],
         f.coalesce(
-            all_metering_points[Colname.energy_supplier_id],
-            production_and_consumption_metering_points[es],
+            all_child_metering_points[Colname.energy_supplier_id],
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:113]: I'm not sure why this is needed?

Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:113]: yeah it is not, it was from before it was split into child and parent. I will change it

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_get_metering_points_and_child_metering_points.py [hunk 6]
```diff
@@ -22,72 +24,384 @@
...
+    def test__returns_two_child_metering_points_with_the_same_period_as_the_parent_metering_points(
+        self,
+        spark: SparkSession,
+    ):
+        """
+        input metering points:
+        parent:  |-----------------|
+        parent:                    |-----------------|
+        child:   |-----------------------------------|
+        output metering points:
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_get_metering_points_and_child_metering_points.py:82]: really nice illustrations!

As you mentioned, you could consider to remove parent on the ouput part

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py [hunk 7]
```diff
@@ -35,61 +35,119 @@ def get_metering_points_and_child_metering_points(
...
-        ).alias(
+    child_metering_points_with_energy_suppliers = all_child_metering_points.join(
+        potential_parent_metering_points,
+        (
+            potential_parent_metering_points[mp]
+            == all_child_metering_points[Colname.parent_metering_point_id]
+        )
+        & (
+            (
+                (
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:61]: "# When child from date is within parent period"

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:61]: Will you add this as a comment?

Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:61]: should be there

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**